An effort to update Readme

OrkoHunter opened the pull request:
Here is an attempt to write the Readme with [About](https://github.com/orkohunter/networkx-metis/tree/readme#what-is-networkx-metis), [Installation](https://github.com/orkohunter/networkx-metis/tree/readme#installation), [Documentation](https://github.com/orkohunter/networkx-metis/tree/readme#documentation) and [Contribution](https://github.com/orkohunter/networkx-metis/tree/readme#contribute-to-networkx-metis) sections. Comments are most welcome.

The repository with updated readme will look similar to the [readme](https://github.com/orkohunter/networkx-metis/tree/readme) branch on my fork.

OrkoHunter commented:
Good work [landscape-bot](https://github.com/landscape-bot) !

debugger22 commented on README.md:
```diff
@@ -1,2 +1,132 @@
+
+[![Build Status](https://travis-ci.org/networkx/networkx-metis.svg?branch=master)](https://travis-ci.org/networkx/networkx-metis)
+[![Code Health](https://landscape.io/github/networkx/networkx-metis/master/landscape.svg?style=flat)](https://landscape.io/github/networkx/networkx-metis/master)
+[![Documentation Status](https://readthedocs.org/projects/networkx-metis/badge/?version=latest)](https://readthedocs.org/projects/networkx-metis/?badge=latest)
```
Change the target to `https://networkx-metis.readthedocs.org/en/latest/`

debugger22 commented on README.md:
````diff
@@ -1,2 +1,132 @@
+######Requirements
+
+```sh
+sudo pip install Cython #cython
````
Remove all `sudo` from the installation commands. One shouldn't use system's Python.

SanketDG commented on README.md:
````diff
@@ -1,2 +1,132 @@
+######NetworkX-Metis
+
+```sh
+sudo pip3 install https://github.com/networkx/networkx-metis/zipball/master
````
I think just adding a line to use `pip3` instead of `pip` when using Python 3 should be enough. This approach is kind of verbose.

chebee7i replied:
Another approach is to use `conda` throughout and then your commands will be the same in each situation, only the `conda create` command would be different.

SanketDG commented on README.md:
```diff
@@ -1,2 +1,132 @@
+
+[NetworkX](https://github.com/networkx/networkx) is a Python package for the creation,
+manipulation and study of the structure, dynamics, and functions of complex networkx.
+[METIS](http://glaros.dtc.umn.edu/gkhome/views/metis) is a C library written for
```
Might wanna change this to http://glaros.dtc.umn.edu/gkhome/metis/metis/overview to be more relevant.

SanketDG commented on README.md:
````diff
@@ -1,2 +1,132 @@
+```sh
+git clone https://github.com/networkx/networkx-metis.git
+cd networkx-metis/
+sudo python setup.py build --compiler=mingw32
````
Since you are just specifying the compiler, you might add the installation instructions of `mingw` since it doesn't come installed by default.

chebee7i commented:
It is up to you if you want to have landscape comment. I think it tends to make the PRs noisy.

OrkoHunter commented:
Yes. Let's disable the comments. I'll check for the errors, smells from the site itself.

MridulS commented:
@chebee7i On a side note: Can we have landscape for networkx also?

OrkoHunter commented:
There's an issue open about it https://github.com/networkx/networkx/issues/1492

ysitu commented on README.md:
```diff
@@ -1,2 +1,98 @@
-# networkx-metis
-NetworkX Addon to allow graph partitioning with METIS
+# NetworkX-Metis
```
Use "NetworkX-METIS" throughout.

ysitu commented on README.md:
```diff
@@ -1,2 +1,98 @@
+ * [Documentation](#documentation)
+ * [Contribute](#contribute-to-networkx-metis)
+
+<a id="about"></a>
```
Do you still need these anchors?

ysitu commented on README.md:
```diff
@@ -1,2 +1,98 @@
+ - [Solve Bugs and Issues](https://github.com/networkx/networkx-metis/issues?page=1&state=open)
+ - Write Tutorials, Examples and Documentation
+
+Note : You should be familiar with Python, C and/or Cython so as to contribute to NetworkX-Metis.
```
This sentence seems unnecessary to me.

ysitu commented on README.md:
```diff
@@ -1,2 +1,98 @@
+[METIS](http://glaros.dtc.umn.edu/gkhome/views/metis) is a C library written for
+partitioning graphs, partitioning finite element meshes, and producing
+fill reducing orderings for sparse matrices. NetworkX-Metis uses Cython to wrap the
+METIS library in order to be used by writting Python code using NetworkX.
```
Change "in order to be used by writting Python code using NetworkX" to "to make it available in Python".

ysitu commented on README.md:
```diff
@@ -1,2 +1,98 @@
+-
+Documentation is available at 
+
+[*http://networkx-metis.readthedocs.org/en/latest*](http://networkx-metis.readthedocs.org/en/latest)
```
Merge with previous paragraph. Add period. Remove italics.

ysitu commented on README.md:
```diff
@@ -1,2 +1,98 @@
+## Windows
+
+The installation for windows has a small trick in addition to what is required for Linux/Mac.
+For Cython to compile the library, a `--compiler=mingw32` flag is needed.
```
"Installation on Windows is largely the same as on Linux/Mac, although for Cython to compile the library, an extra --compiler=mingw32 flag is needed."

OrkoHunter commented:
Addressed all comments.

ysitu commented on README.md:
````diff
@@ -1,2 +1,88 @@
+
+```sh
+pip install Cython #cython
+pip install https://github.com/networkx/networkx/zipball/master #networkx
````
We should be able to do `pip install networkx` after networkx-1.10 is released with addon support.

ysitu commented on README.md:
````diff
@@ -1,2 +1,88 @@
+######NetworkX-METIS
+
+```sh
+pip install https://github.com/networkx/networkx-metis/zipball/master
````
`pip install networkx-metis`? We will release this to PyPI.

ysitu commented on README.md:
````diff
@@ -1,2 +1,88 @@
+```sh
+pip install Cython #cython
+pip install https://github.com/networkx/networkx/zipball/master #networkx
+```
````
We can merge the two "Requirements" sections given that they are the same.

ysitu commented on README.md:
```diff
@@ -1,2 +1,88 @@
+
+###Install with pip
+
+#####Using pip
```
Remove.

ysitu commented on README.md:
```diff
@@ -1,2 +1,88 @@
+-
+## Linux/Mac
+
+###Install with pip
```
"From PyPI".

ysitu commented on README.md:
````diff
@@ -1,2 +1,88 @@
+pip install https://github.com/networkx/networkx-metis/zipball/master
+```
+
+###Install using git
````
"From GitHub"

ysitu commented on README.md:
```diff
@@ -1,2 +1,88 @@
+## Windows
+
+Installation on Windows is largely the same as on Linux/Mac, although for Cython
+to compile the library, an extra --compiler=mingw32 flag is needed.
```
Put "--compiler=mingw32" in backticks?

ysitu commented on README.md:
````diff
@@ -1,2 +1,88 @@
+python setup.py install
+```
+
+provided that Cython and NetworkX has been installed as described in above sections.
````
Change "has been" to "have been".

MridulS commented:
On a side note: Should we advise users to use virtualenv? I don't think running `pip install networkx` will work. It has to be `sudo pip install networkx` or to avoid touching OS files  `pip install --user networkx`, right?

OrkoHunter commented:
This is the general way to show the installation procedure. You do not add `sudo` anywhere, nor you are advised to touch the system files.

MridulS commented:
Should we add a note or leave it to the user to figure it out? (This can happen with someone who is new to python)

ysitu commented on README.md:
```diff
@@ -1,2 +1,75 @@
+-
+
+For a detailed summary of all the coding guidelines and development workflow, please visit
+the [Developer Guide](http://networkx.github.io/documentation/networkx-1.9.1/developer/index.html).
```
Point to the latest version on readthedocs.

ysitu commented:
@MridulS I think that knowledge about `pip` is not something for us to worry about. A user who is to use `pip install` should know what it does. `networkx-metis` will not be the only package he/she runs into problems with.

ysitu commented on README.md:
````diff
@@ -1,2 +1,75 @@
+
+```sh
+git clone https://github.com/networkx/networkx-metis.git
+cd networkx-metis/
````
Windows uses backslashes.

OrkoHunter replied:
Removed

ysitu commented on README.md:
```diff
@@ -1,2 +1,75 @@
+## Windows
+
+Installation on Windows is largely the same as on Linux/Mac, although for Cython
+to compile the library, an extra `--compiler=mingw32` flag is needed.
```
On second thought, what makes MinGW necessary? Or why does not Visual C++ work?

OrkoHunter replied:
`MinGW` is a recommendation of cython. They have it all over places. I didn't really try any other compiler though.

ysitu replied:
https://github.com/cython/cython/wiki/CythonExtensionsOnWindows is worth reading.

OrkoHunter replied:
Enlightening. I liked this line very much - Mingw is a gcc compiler. On windows, always use the stuff provided by Microsoft.
I say, we add the relevant part here or we can put some of the information and point to the page.

OrkoHunter commented:
I've added few lines in the `Windows` section for installation. Also, I've added an example, though I'm sure we can have (a) better example(s). Can anyone suggest a good graph to partition to put as an example?

ysitu commented on README.md:
````diff
@@ -1,2 +1,90 @@
+
+```sh
+pip install Cython #cython
+pip install networkx #networkx
````
Adapt for #36.

ysitu commented on README.md:
````diff
@@ -1,2 +1,90 @@
+###From PyPI
+
+```sh
+pip install networkx-metis
````
Adapt for #36.

OrkoHunter replied:
Did you mean `pip install nxmetis` ? setup.py still has the name `networkx-metis`

ysitu replied:
This is about branding of the package. The module name may be `nxmetis`, but `networkx-metis` sounds better as a package name. So let's keep it.

OrkoHunter replied:
Yes. +1

ysitu commented on README.md:
```diff
@@ -1,2 +1,90 @@
+In [3]: G = nx.complete_graph(10)
+
+In [4]: metis.partition(G, 2)
+Out[4]: (25, [[0, 1, 2, 3, 6], [4, 5, 7, 8, 9]])
```
Make this look like interactive interpreter output. Not everyone uses IPython.

OrkoHunter replied:
Done

SanketDG commented on README.md:
```diff
@@ -1,2 +1,90 @@
+-
+
+For a detailed summary of all the coding guidelines and development workflow, please visit
+the [Developer Guide](https://networkx.readthedocs.org/en/latest/developer/index.html).
```
How will this fit into networkx-metis? I think we should have some separate pages as a dev guide over at networkx-metis' docs, a much shorter version of what is at networkx.

OrkoHunter replied:
These are a very good resource of git guide. We can only replicate them with replacing networkx to networkx-metis. I don't see anything there to cut out.

ysitu replied:
I think that it is reasonable to point to documentation of NetworkX. This is *networkx*-metis after all.
